tee-worker/omni-executor/rpc-server/src/methods/omni/get_hyperliquid_signature_data.rs

```diff
@@ -0,0 +1,508 @@
+		digest_input[0] = 0x19;
+		digest_input[1] = 0x01;
+		digest_input[2..34].copy_from_slice(&self.domain().hash_struct()[..]);
+		digest_input[34..66].copy_from_slice(&self.struct_hash()[..]);
+		keccak256(digest_input)
+	}
+}
+
+impl HyperliquidEip712Signature for ApproveAgentAction {
+	fn domain(&self) -> Eip712Domain {
+		eip712_domain! {
+			name: "HyperliquidSignTransaction",
+			version: "1",
+			chain_id: self.signature_chain_id,
+			verifying_contract: Address::ZERO,
+		}
+	}
+
+	fn struct_hash(&self) -> B256 {
+		let items = (
+		keccak256("HyperliquidTransaction:ApproveAgent(string hyperliquidChain,address agentAddress,string agentName,uint64 nonce)"),
+		keccak256(&self.hyperliquid_chain),
+		&self.agent_address,
+		keccak256(self.agent_name.as_deref().unwrap_or("")),
+		&self.nonce
+	);
+		keccak256(items.abi_encode())
+	}
+}
+
+impl HyperliquidEip712Signature for Withdraw3Action {
+	fn domain(&self) -> Eip712Domain {
+		eip712_domain! {
+			name: "HyperliquidSignTransaction",
+			version: "1",
+			chain_id: self.signature_chain_id,
+			verifying_contract: Address::ZERO,
+		}
+	}
+
+	fn struct_hash(&self) -> B256 {
+		let items = (
+		keccak256("HyperliquidTransaction:Withdraw3(string hyperliquidChain,string amount,uint64 time,address destination)"),
+		keccak256(&self.hyperliquid_chain),
+		keccak256(&self.amount),
+		&self.time,
+		&self.destination
+	);
+		keccak256(items.abi_encode())
+	}
+}
+
+impl HyperliquidEip712Signature for ApproveBuilderFeeAction {
+	fn domain(&self) -> Eip712Domain {
+		eip712_domain! {
+			name: "HyperliquidSignTransaction",
+			version: "1",
+			chain_id: self.signature_chain_id,
+			verifying_contract: Address::ZERO,
+		}
+	}
+
+	fn struct_hash(&self) -> B256 {
+		let items = (
+		keccak256("HyperliquidTransaction:ApproveBuilderFee(string hyperliquidChain,string maxFeeRate,address builder,uint64 nonce)"),
+		keccak256(&self.hyperliquid_chain),
+		keccak256(&self.max_fee_rate),
+		&self.builder,
+		&self.nonce
+	);
+		keccak256(items.abi_encode())
+	}
+}
+
+pub fn register_get_hyperliquid_signature_data(module: &mut RpcModule<RpcContext>) {
+	module
+		.register_async_method("omni_getHyperliquidSignatureData", |params, ctx, _| async move {
+			let params = params.parse::<GetHyperliquidSignatureDataParams>().map_err(|e| {
+				error!("Failed to parse params: {:?}", e);
+				ErrorCode::ParseError
+			})?;
+
+			let verification_code = match &params.user_auth {
+				UserAuth::Email(code) => code.clone(),
+				_ => {
+					error!("Only email authentication is supported for this method");
+					return Err(ErrorObject::from(ErrorCode::InvalidParams));
+				},
+			};
+
+			let email = match &params.user_id {
+				UserId::Email(email) => email.clone(),
+				_ => {
+					error!("User ID must be an email for this method");
+					return Err(ErrorObject::from(ErrorCode::ParseError));
+				},
+			};
+
+			verify_email_authentication(ctx.clone(), &params.client_id, &email, &verification_code)
```

Comment:
See the old comment - is email code required for every action, or can it be done by sig? Need confirmation from backend